src/Plugin/EntityBrowser/Display/Modal.php

```diff
@@ -164,10 +178,53 @@ public function displayEntityBrowser() {
   }
 
   /**
+   * Generates the content and opens the modal.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   *
+   * @return \Drupal\Core\Ajax\AjaxResponse
+   *   An ajax response.
+   */
+  public function openModal(array &$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    $parents = $triggering_element['#parents'];
+    array_pop($parents);
+    $parents = array_merge($parents, ['path']);
+
+    // Trying to get the path value from $form_state returns null. I am not sure
+    // why, but $form_state->storage does't have values for our field.
+    // @todo try setting the storage? $form_state->setStorage($storage);
```

Comment:
This probably happens since parent form wasn't able to detect this elements and removed user values as a result (Drupal does that for security reasons.).

I assume you are testing this with entity reference widget. We can try to move this one level up and see if that fixes the issue.